rubix-spi/src/main/java/com/qubole/rubix/spi/CacheUtil.java

```diff
@@ -152,12 +152,13 @@ public static String getDirPath(int dirIndex, Configuration conf)
    *
    * @param remotePath  The path for a remote location.
    * @param conf        The current Hadoop configuration.
+   * @param generationNumber the generation number for local file corresponding to remote location
    * @return The local path location.
    */
-  public static String getLocalPath(String remotePath, Configuration conf)
+  public static String getLocalPath(String remotePath, Configuration conf, int generationNumber)
   {
     final String absLocation = getDirectory(remotePath, conf);
-    return absLocation + "/" + getName(remotePath);
+    return absLocation + "/" + getName(remotePath) + "g" + generationNumber;
```

Comment:
use string.format for better readability.

Reply:
And add a "_" before "g"